In [2]:
import numpy as np
import scipy.stats as st
a = np.array([178, 184, 149, 193, 186, 173, 169, 175, 159 ,174])

In [4]:
x1 = np.mean(a)
x1

174.0

In [5]:
D1 = np.var(a, ddof=1)
D1

166.44444444444446

In [8]:
n  = 10
alpha = 0.05
t1 = st.t.ppf(1-alpha/2, n - 1) #t1 = st.t.ppf(0.975, 9) по таблице поиск на пересечении 0.975(альфа пополам и минус 100%) и 9(степени свободы)
t1

2.2621571627409915

In [9]:
x1 - t1 * np.sqrt(D1 / n)

164.77094092110704

In [10]:
x1 + t1 * np.sqrt(D1 / n)

183.22905907889296

[164.77 ; 183.22] доверительный интервал - с вероятностью 95% среднее занчение ГС попадет в интервал

Оценить различие в росте между двумя средними арифметическими популяции А и В с помощью 95% доверительного интервала

delta ± t@alpha/2 * S@delta

D = (D1 + D2) / 2 - Дисперсия

S@delta = sqrt (D / n1 + D / n2)

In [12]:
a = np.array([178, 184, 149, 193, 186, 173, 169, 175, 159, 174])
b = np.array([150, 154, 167, 165, 171, 150, 158, 170 ,175, 160])

print(len(a), len(b))

10 10


In [13]:
x1 = np.mean(a)
x2 = np.mean(b)
delta = x1 - x2
print(x1, x2, delta)

174.0 162.0 12.0


In [16]:
D1 = np.var(a, ddof=1) # несмещеная дисперсия для выборки 1
D2 = np.var(b, ddof=1) # несмещеная дисперсия для выборки 2
D = (D1 + D2) / 2      # объединенная оценка дисперсии

print(D1, D2, D)

166.44444444444446 80.0 123.22222222222223


In [18]:
SE = np.sqrt(D/10 + D/10) #Стандартная ошибка разницы средних
SE

4.964317117635058

In [21]:
t = st.t.ppf(0.975, 18) #степени свободы = 2*(n-1)=18, (1 - alpha = 0.05 / 2) = 0.975
t

2.10092204024096

In [24]:
L = delta - t * SE # нижняя граница
U = delta + t * SE # верхняя граница
print(L)
print([L,U])

1.570356752815032
[1.570356752815032, 22.42964324718497]
